core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java

```diff
@@ -231,7 +231,7 @@ private static Type getActualReturnType(Method method) {
         final Class<?> returnType;
         final Type genericReturnType;
 
-        if (KotlinUtil.isSuspendingFunction(method)) {
+        if (KotlinUtil.isSuspendingFunction(method) && !KotlinUtil.isReturnTypeNothing(method)) {
```

Comment:
I thought this workaround can be solution for this issue, but if there is any better idea, please advice me freely 🙇

Reply:
The workaround looks good to me.
@ks-yim Could you review this PR?

Reply:
Looks good to me, too but we should do the same thing at `AnnotatedDocServicePlugin.java:L159 `.

BTW, I've just found that `DocService` shows `Unit` return type correctly for suspending functions while it shows `void` for normal kotlin functions 😂..

Maybe we can introduce `isKFunction` method and use it instead of `isSuspendingFunction` here and `AnnotatedDocServicePlugin` to show the correct return type? (if it worth the effort, though)

Reply:
Thank you for suggestion, @ks-yim :) Do you mean the method like it?

```kotlin
// ArmeriaKotlinUtil.kt
internal fun isKFunction(method: Method): Boolean {
    return method.kotlinFunction != null && !isReturnTypeNothing(method)
}
```

Reply:
@jupiny 
Just...:
```kotlin
// ArmeriaKotlinUtil.kt
internal fun isKFunction(method: Method): Boolean {
    return method.kotlinFunction != null
}
```
and use it instead of `isSuspendingFunction` leaving `isReturnTypeNothing` as it currently is.

Reply:
Ah, actually I tried so, but `Nothing` method is still considered as `kFunction`)(`method.kotlinFunction` is not null). So it makes the same problem. 🤔
If my understand is wrong, please let me know. 😭

Reply:
@jupiny Sorry to confuse you.

What I meant was it would be better to do the following to get the correct return type though it's a bit out of the scope of this PR.. 😅;;

...at `AnnotatedService.java:L234`:
```kotlin
if (KotlinUtil.isKFunction(method) && !KotlinUtil.isReturnTypeNothing(method)) {
    // Use kotlin reflection
} else {
    // Use Java reflection
}
```

...and this at `AnnotatedDocServicePlugin:L158`:
```kotlin
final TypeSignature returnTypeSignature =
        isKFunction(method) ? toTypeSignature(kFunctionGenericReturnType(method))
                            : toTypeSignature(method.getGenericReturnType());
```

Reply:
Aha, I misunderstand a little. Updated following your comments!